### DISCUSSION TITLE
MX Check on From: header

### DISCUSSION THREAD

**Discussion by dblas** [rspamd/rspamd#3703] [Ideas]
Hello to all and thank you for this well-made Rspamd product.
There are plenty of features and in Rspamd but It seems that one of them is missing:
the capacity to check the existence of a mail domain that lies into the From: header (I'm not speaking of SMTP From envelope).
More and more attackers use this lack of checking to forge inexistant domains that are similar to the ones they are targeting.
The SMTP From: mail domain that is used in this kind of attack exists, of course but the recipient doesn't see it, never.

And the From: header (the one that the recipient car easily analyse) can then be forced to @goeqle.com for example a domain that doesn't exist but which is very similar to google.com, visually.

There may be a way to check an MX coming from that header using LUA. What do you thing of it?
Thanx again.
db

**Comment by defulmere** [comment]
I deal with fake domains in Postfix before the mail ever gets to rspamd.

Here's an example from the [Postfix docs](http://postfix.cs.utah.edu/SMTPD_ACCESS_README.html):

```
# Don't accept mail from domains that don't exist.
smtpd_sender_restrictions = reject_unknown_sender_domain
```

Not sure if you're using Postfix but maybe your MTA has something similar.
